### DISCUSSION TITLE
Replicating sextractor command with sep.

### DISCUSSION THREAD

**Discussion by igorkolesnikov13** [sep-developers/sep#113] [Q&A]
Hi to everyone.

Guys, I'm really looking forward to being able to improve my code with sep.
I have the following command that I'm currently employing in my pipeline:
`sextractor image.fit -BACK_SIZE 32 -DETECT_THRESH 0.8 -CATALOG_NAME image.cat  -CHECKIMAGE_TYPE SEGMENTATION,BACKGROUND -CHECKIMAGE_NAME image_segmented.fits, image_background.fits -VERBOSE_TYPE QUIET`

I already found the part of the `Background` image in the documentation, but the part of the `Segmented` image and `.cat` file (with several parameters) is not clear. Is this possible to achieve with sep?
Appreciate any help.

**Comment by kbarbary** [comment]
SEP breaks what `sextractor` does into several different functions:

- background subtraction: `sep.Background`
- object detection: `sep.extract`
- aperture photometry: `sep.sum_circle`, `sep.sum_ellipse`, etc.

The [tutorial](https://sep.readthedocs.io/en/latest/tutorial.html) gives a good overview of how to piece these together.

Specifically, for some of the options you're using:

- `-BACK_SIZE 32`: This is an option in `sep.Background`
- `DETECT_THRESH 0.8`: This is an option in `sep.extract`. For example: `sep.extract(data_sub, 0.8, err=bkg.globalrms)`
- `-CATALOG_NAME image.cat`: This is not relevant because `SEP` doesn't deal with reading or writing files. Instead, some of this information is included in the structured numpy array returned by `sep.extract`.
-  `-CHECKIMAGE_TYPE SEGMENTATION,BACKGROUND -CHECKIMAGE_NAME image_segmented.fits, image_background.fits`: The tutorial shows how to get a background image. The segmentation map can be returned with the `segmentation_map=True` argument to `sep.extract`.
